File: Source/Magazine.Web/src/app/events/article/article.service.event.ts
import { Injectable } from "@angular/core";
import { EntityServiceEvent } from "../../contracts/event/entity.service.event";
import { ArticleService } from "src/app/services/article/article.service";
import { jsonProperty } from "ts-serializable";

/** Событие сервиса сущности "Статья". */
@Injectable({
    providedIn: 'root'
})
export abstract class ArticleServiceEvent extends EntityServiceEvent {

    @jsonProperty()
    /** Идентификатор статьи. */
    protected id: number;

    constructor(className: string,
                id: number) {
                    debugger;
        super(className);
        this.id = id;
    }

    abstract visit(service: ArticleService): void;
}

File: Source/Magazine.API/Contracts/Event/ArticleServiceEvent.cs
using Newtonsoft.Json;

namespace Magazine.API.Contracts.Event
{
    /// <summary>
    /// Событие сервиса сущности "Статья".
    /// </summary>
    public abstract class ArticleServiceEvent : EntityServiceEvent
    {
        [JsonProperty]
        /// <summary>
        /// Идентификатор статьи.
        /// </summary>
        protected int Id { get; set; }
    }
}
